// database/dbConnect.js
var mysql=require('mysql');
function connectServer(){
    var client=mysql.createConnection({
        host:'localhost',
        user:'root',
        password:'',
        database:'gongdan'
    });
    return client;
}
/*用户登录*/
function  selectUser(client,username,password,callback){
    client.query('SELECT * FROM `serve` WHERE `s_name`="'+username+'" and `s_pwd`="'+password+'"',function(err,results,fields){
        if(err) throw err;

        callback(results);
    });
}

/*查找所有的员工的数据*/
function  selectKeyEmFun(client,keyword,callback){
    client.query("SELECT * FROM `serve` WHERE `s_user` LIKE ?",
        [keyword],
        function(err,results,fields){
            if(err) throw err;
            callback(results);

        });
}
/*查找某个员工组的员工的数据*/
function  selectEmIDFun(client,id,callback){
    client.query("SELECT * FROM `serve` WHERE `sg_id`=?",
        [id],
        function(err,results,fields){
            if(err) throw err;
            callback(results);

        });
}
/*查找员工组的数据*/
function  selectEmgroupFun(client,callback){
    client.query("SELECT * FROM `servegroup`",
        function(err,results,fields){
            if(err) throw err;
            callback(results);

        });
}
//查找员工  所有的数据
function  selectEmployeeFun(client,callback){
    client.query("SELECT * FROM `serve` sv,`servegroup` sp WHERE sv.`sg_id`=sp.`sg_id`",
        function(err,results,fields){
            if(err) throw err;
            callback(results);

        });
}
//查找员工  所有的数据
function  selectEmployeeForPageFun(client,currentPage,callback){/*   currentPage =>当前页数    */
    var pageSql="SELECT *,(SELECT count(*) FROM `serve` sv,`servegroup` sp WHERE sv.`sg_id`=sp.`sg_id`) as count FROM `serve` sv,`servegroup` sp WHERE sv.`sg_id`=sp.`sg_id` limit ?,2";
    client.query(pageSql,[2*currentPage],
        function(err,results,fields){
            if(err) throw err;
            callback(results);

        });
}
//查找某个员工的数据
function  selectEmployeeIDFun(client,id,callback){
    client.query("SELECT * FROM `serve` sv,`servegroup` sp WHERE sv.`sg_id`=sp.`sg_id` and `s_id`=?",
        [id],
        function(err,results,fields){
            if(err) throw err;
            callback(results);

        });
}
//新增员工的数据
function insertEmployeeFun(client,name,user,password,group,part,phone,nickname,email,headPic,callback){

    client.query("INSERT INTO `serve`(`s_id`, `s_name`, `s_user`, `s_pwd`, `sg_id`, `s_part`, `s_phone`, `s_nickname`, `s_email`, `s_headPic`) VALUES (NULL,?,?,?,?,?,?,?,?,?)",
        [name,user,password,group,part,phone,nickname,email,headPic],
        function(err,results){
            if( err ){
                console.log( "error:" + err.message);
                return err;
            }else{
                callback(err);
            }
        });
}
//修改员工的数据
function updateEmployeeIdFun(client,name,group,part,phone,nickname,headPic,id,callback){
    client.query("UPDATE `serve` SET `s_name`=?,`sg_id`=?,`s_part`=?,`s_phone`=?,`s_nickname`=?,`s_headPic`=? WHERE `s_id`=?",
        [name,group,part,phone,nickname,headPic,id],
        function(err,results){
            if( err ){
                console.log( "error:" + err.message);
                return err;
            }else{
                console.log(name+"========="+group+"========="+part+"========="+phone+"========="+nickname+"========="+headPic+"========="+id);
                callback(err);
            }
        });
}
//删除某条员工的数据
function  delEmployeeIDFun(client,id,callback){
    client.query('DELETE FROM `serve` WHERE `s_id`="'+id+'"',function(err,results,fields){
        if(err) throw err;
        callback(results);
    });
}

/*查找所有 显示工单列表*/

function  selectFun(client,callback){
//    DATE_FORMAT(time,'%Y-%m-%d')   转化成format格式
    client.query("SELECT distinct *, DATE_FORMAT(o_starttime,'%Y-%m-%d') as datetime FROM `orders` os,`customer` cu,`serve` se,`servegroup` sp WHERE os.`cu_id`=cu.`cu_id` and os.`s_id`=se.`s_id` and se.`sg_id`=sp.`sg_id`",
        function(err,results,fields){
            if(err) throw err;
            callback(results);

        });
}
/*查找所有 分页显示工单列表*/

function  selectForPagesFun(client,currentPage,callback){
//    currentPage =>当前页数
    var pageSql="SELECT distinct *,DATE_FORMAT(o_starttime,'%Y-%m-%d') as datetime,(SELECT count(*) FROM `orders` os,`serve` se where os.`s_id`=se.`s_id`) as count FROM `orders` os,`customer` cu,`serve` se,`servegroup` sp " +
        "WHERE os.`cu_id`=cu.`cu_id` and os.`s_id`=se.`s_id` and se.`sg_id`=sp.`sg_id` limit ?,2";
    client.query(pageSql,[2*currentPage],
        function(err,results,fields){
            if(err) throw err;
            callback(results);
        });
}
//查找ID的工单 显示工单详情
function  selectIdFun(client,id,callback){
//    DATE_FORMAT(time,'%Y-%m-%d')   转化成format格式
    client.query("SELECT distinct *, DATE_FORMAT(o_starttime,'%Y-%m-%d') as starttime FROM `orders` os,`customer` cu,`serve` se,`servegroup` sp WHERE os.`cu_id`=cu.`cu_id` and os.`s_id`=se.`s_id` and se.`sg_id`=sp.`sg_id` and `o_id`=?",
        [id],
        function(err,results,fields){
            if(err) throw err;
            callback(results);

        });
}
//新增工单
function insertFun(client,title,content,status,level,tags,starttime,follower,remark,qudao,upload,customer,serve,userID,callback){
    client.query("INSERT INTO `orders`(`o_id`, `o_title`, `o_description`, `o_status`, `o_level`, `o_tags`, `o_starttime`, `o_follower`, `o_remark`, `o_qudao`, `o_upload`, `cu_id`, `s_id`,`cr_id`)" +
            " VALUES (NULL,?,?,?,?,?,?,?,?,?,?,?,?,?)",
        [title,content,status,level,tags,starttime,follower,remark,qudao,upload,customer,serve,userID],
        function(err,results){
            if( err ){
                console.log( "error:" + err.message);
                return err;
            }else{
                callback(err);
            }
        });
}
//工单转移  修改员工和工单备注
function updateEmFun(client,remark,emplo,id,callback){
    client.query("UPDATE `orders` SET `o_remark`=?,`s_id`=? WHERE `o_id`=?",
        [remark,emplo,id],
        function(err,results){
            if( err ){
                return err;
            }else{
                callback(err);
            }
        });
}

//工单修改（某个ID）
function updateIdFun(client,title,content,status,level,tags,follower,remark,upload,customer,sarve,id,callback){
    client.query("UPDATE `orders` SET `o_title`=?,`o_description`=?,`o_status`=?,`o_level`=?,`o_tags`=?," +
            "`o_follower`=?,`o_remark`=?,`o_upload`=?,`cu_id`=?,`s_id`=? WHERE `o_id`=?",
        [title,content,status,level,tags,follower,remark,upload,customer,sarve,id],
        function(err,results){
            if( err ){
                return err;
            }else{
                callback(err);
            }
        });
}
//工单删除（某个ID）
function delIdFun(client,id,callback){
    client.query('DELETE FROM `orders` WHERE `o_id`="'+id+'"',function(err,results,fields){
        if(err) throw err;
        callback(results);
    });
}
//------------------------------------------------------------------- 员工组 ----------------------------------------------------------------------------------
//查询所有的员工组数据
function selectGroupFun(client,callback){
    client.query("SELECT *,DATE_FORMAT(`sg_startdate`,'%Y-%m-%d') as datetime FROM `servegroup`",
        function(err,results,fields){
            if(err) throw err;
            callback(results);

        });
}
//查询某个ID的员工组的数据
function selectGroupIdFun(client,id,callback){
    client.query("SELECT *,DATE_FORMAT(`sg_startdate`,'%Y-%m-%d') as datetime FROM `servegroup` where `sg_id`=?",[id],
        function(err,results,fields){
            if(err) throw err;
            callback(results);

        });
}
//新增员工组的数据
function insertGroupFun(client,name,serve,date,callback){
    client.query("INSERT INTO `servegroup`(`sg_id`, `sg_name`, `sg_serves`, `sg_startdate`) VALUES (NULL,?,?,?)",
        [name,serve,date],
        function(err,results){
            if( err ){
                console.log( "error:" + err.message);
                return err;
            }else{
                callback(err);
            }
        });
}
//修改员工组的数据
function updateGroupIdFun(client,title,id,callback){
    client.query("UPDATE `servegroup` SET `sg_name`=? WHERE `sg_id`=?",
        [title,id],
        function(err,results){
            if( err ){
                return err;
            }else{
                callback(err);
            }
        });
}
//删除某个ID的员工组的数据
function delGroupIdFun(client,id,callback){
    client.query('DELETE FROM `servegroup` WHERE `sg_id`="'+id+'"',function(err,results,fields){
        if(err) throw err;
        callback(results);
    });
}
//------------------------------------------------------------------- 个人中心 ----------------------------------------------------------------------------------
function selectPersonFun(client,id,callback){
    client.query("SELECT * FROM `serve` WHERE `s_id`=?",[id],
        function(err,results,fields){
            if(err) throw err;
            callback(results);
        });
}
function updatePersonFun(client,tel,headPic,id,callback){
    client.query("UPDATE `serve` SET `s_tel`=?,`s_headPic`=? WHERE `s_id`=?",
        [tel,headPic,id],
        function(err,results){
            if( err ){
                return err;
            }else{
                callback(err);
            }
        });
}
function updatePwdFun(client,password,id,callback){
    client.query("UPDATE `serve` SET `s_pwd`=? WHERE `s_id`=?",
        [password,id],
        function(err,results){
            if( err ){
                return err;
            }else{
                callback(err);
            }
        });
}
/*查询所有客户的数据*/

function  selectCustomFun(client,callback){
    client.query("SELECT * FROM `customer`cu,`company` cp WHERE cu.`c_id` =cp.`c_id`",
        function(err,results,fields){
            if(err) throw err;
            callback(results);

        });
}
/*查询某个客户的数据*/

function  selectCustomIDFun(client,id,callback){
    client.query("SELECT * FROM `customer`cu,`company` cp WHERE cu.`c_id` =cp.`c_id` and cu.`cu_id`=?",
        [id],
        function(err,results,fields){
            if(err) throw err;
            callback(results);

        });
}
//查找客户 通过关键字
function  selectKeywordFun(client,keyword,callback){
    client.query("SELECT * FROM `customer` cu,`company` cp WHERE cu.`c_id`=cp.`c_id` and `cu_name` LIKE ?",
        [keyword],
        function(err,results,fields){
            if(err) throw err;
            callback(results);

        });
}

exports.connect = connectServer;/*创建数据库*/

exports.selectUser  = selectUser;/*用户登录*/

exports.selectCustomFun  = selectCustomFun;/*客户所有的数据信息*/
exports.selectCustomIDFun  = selectCustomIDFun;/*某位客户的数据信息*/
exports.selectKeyEmFun  = selectKeyEmFun;/*通过关键字查询员工的数据*/
exports.selectEmIDFun  = selectEmIDFun;/*某个员工的数据*/

exports.selectEmgroupFun  = selectEmgroupFun;/*员工组的数据*/

exports.selectEmployeeFun  = selectEmployeeFun;/*查询员工数据*/
exports.selectEmployeeForPageFun  = selectEmployeeForPageFun;/*分页查询员工数据*/
exports.selectEmployeeIDFun  = selectEmployeeIDFun;/*查询某位员工数据*/
exports.insertEmployeeFun  = insertEmployeeFun;/*新增员工数据*/
exports.updateEmployeeIdFun  = updateEmployeeIdFun;/*修改员工数据*/
exports.delEmployeeIDFun  = delEmployeeIDFun;/*删除某个员工数据*/

exports.updateEmFun  = updateEmFun;/*转移工单 修改员工和备注*/
exports.selectFun  = selectFun;/*查找工单全部数据*/
 exports.selectForPagesFun  = selectForPagesFun;/*查找工单分页显示全部数据*/
exports.selectIdFun  = selectIdFun;/*查找某个Id的工单数据*/
exports.selectKeywordFun  = selectKeywordFun;/*通过关键字查找客户名称*/
exports.insertFun  = insertFun;/*新增工单*/
exports.updateIdFun  = updateIdFun;/*修改工单*/
exports.delIdFun  = delIdFun;/*删除工单*/

exports.selectGroupFun  = selectGroupFun;/*查询所有的员工组的内容*/
exports.selectGroupIdFun  = selectGroupIdFun;/*查询某个ID的员工组的内容*/
exports.insertGroupFun  = insertGroupFun;/*新增员工组的内容*/
exports.updateGroupIdFun  = updateGroupIdFun;/*修改员工组的内容*/
exports.delGroupIdFun  = delGroupIdFun;/*删除某个ID的员工组的内容*/

exports.selectPersonFun=selectPersonFun;/*个人中心查询信息*/
exports.updatePersonFun=updatePersonFun;/*个人中心修改信息*/
exports.updatePwdFun=updatePwdFun;/*个人中心修改密码*/

//exports.selectUserOrder  = selectUserOrder;/*排序 通过用户 查找*/
//exports.selectStatus  = selectStatus;/*状态查找*/
//exports.selectStaFun  = selectStaFun;/*根据状态查询数据*/
//exports.updateIdFun  = updateIdFun;/*删除某条工单*/
